api/host.py

```diff
@@ -285,13 +285,7 @@ def delete_by_id(host_id_list):
 
     hosts_to_delete = []
     for host in query.all():
-        try:
-            hosts_to_delete.append(host)
-        except sqlalchemy.orm.exc.ObjectDeletedError:
-            logger.exception(
-                "Encountered sqlalchemy.orm.exc.ObjectDeletedError exception during delete_by_id operation.  Host was "
-                "already deleted."
-            )
+        hosts_to_delete.append(host)
```

Comment:
`query.all()` actually returns a pure list, so we can do `hosts_to_delete = query.all()` directly. But as I noted in the next comment, we are not checking now in any way whether the host has been already deleted or not.

I guess we need to figure out how to eagerly load the host here, essentially doing manually what happens on the attribute access. I would not eagerly load all of them at once, just do something like object identity check here.

Reply:
If we’re not going to eagerly load in this PR, please abbreviate this to

```suggestion
    hosts_to_delete = query.all()
```